src/algebra/ordered_monoid.lean

```diff
@@ -903,50 +897,76 @@ lemma min_mul_max [linear_order α] [comm_semigroup α] (n m : α) :
   min n m * max n m = n * m :=
 fn_min_mul_fn_max id n m
 
-section linear_ordered_cancel_add_comm_monoid
-variables [linear_ordered_cancel_add_comm_monoid α]
+/-- A decidable linearly ordered cancellative commutative monoid
+is a commutative monoid with a linear order
+in which multiplication is cancellative and strictly monotone. -/
+@[protect_proj] class linear_ordered_cancel_comm_monoid (α : Type u)
+  extends ordered_cancel_comm_monoid α, linear_order α
+
+/-- A decidable linearly ordered cancellative additive commutative monoid
+is an additive commutative monoid with a decidable linear order
+in which addition is cancellative and strictly monotone. -/
+@[protect_proj] class linear_ordered_cancel_add_comm_monoid (α : Type u)
+  extends ordered_cancel_add_comm_monoid α, linear_order α
+
+attribute [to_additive] linear_ordered_cancel_comm_monoid
+  linear_ordered_cancel_comm_monoid.to_ordered_cancel_comm_monoid
 
-lemma min_add_add_left (a b c : α) : min (a + b) (a + c) = a + min b c :=
-(monotone_id.const_add a).map_min.symm
+section linear_ordered_cancel_comm_monoid
 
-lemma min_add_add_right (a b c : α) : min (a + c) (b + c) = min a b + c :=
-(monotone_id.add_const c).map_min.symm
+variables [linear_ordered_cancel_comm_monoid α]
 
-lemma max_add_add_left (a b c : α) : max (a + b) (a + c) = a + max b c :=
-(monotone_id.const_add a).map_max.symm
+@[to_additive] lemma min_mul_mul_left (a b c : α) : min (a * b) (a * c) = a * min b c :=
+(monotone_id.const_mul' a).map_min.symm
 
-lemma max_add_add_right (a b c : α) : max (a + c) (b + c) = max a b + c :=
-(monotone_id.add_const c).map_max.symm
+@[to_additive]
+lemma min_mul_mul_right (a b c : α) : min (a * c) (b * c) = min a b * c :=
+(monotone_id.mul_const' c).map_min.symm
```

Comment:
Why have these proofs changed to use the `'` version of the lemmas?

```suggestion
(monotone_id.mul_const c).map_min.symm
```

Reply:
The doc string at the top of the file is relevant here.

```markdown
## Implementation details

Unfortunately, the number of `'` appended to lemmas in this file
may differ between the multiplicative and the additive version of a lemma.
The reason is that we did not want to change existing names in the library.
```

`monotone.mul_const` is from `algebra.ordered_ring` and requires `linear_ordered_semiring α`, whereas `monotone.mul_const'` is from this file and requires `ordered_comm_monoid α`, and `monotone.mul_const'` is the multiplicative version of `monotone.add_const`

Reply:
See https://github.com/leanprover-community/mathlib/blob/d12a7319934516253eb85a7f90cc47fc20621cdb/src/algebra/ordered_monoid.lean#L190-L191 for reference. The additive and multiplicative versions of this lemma have a different number of `'`s.

Reply:
Eric you're just making suggestions that don't compile. That function doesn't even exist (or at least it's not imported at this time). You can ask the author of
```
@[to_additive monotone.add_const]
lemma monotone.mul_const' (hf : monotone f) (a : α) : monotone (λ x, f x * a) :=
hf.mul' monotone_const
```
why they called it `mul_const'` and not `mul_const` (it will be something to do with `mul_const` being taken up by some ring lemma probably, this is usually the reason for having more mul lemmas than add lemmas) but given that it's there in mathlib and the to_additive database explicitly identifies them, my code seems to be correct. You can grep through mathlib and see tons of examples of where `to_additive` has been given an explicit name for the additive version because the algorithm to change names has failed and the issue is usually that mul lemmas might have a ring version and a group version (and a group isn't a ring and a ring isn't a group, so you can't somehow "fix" this easily)

Reply:
> Eric you're just making suggestions that don't compile

Sorry about that!

Reply:
```
lemma monotone.mul_const (hf : monotone f) (ha : 0 ≤ a) :
  monotone (λ x, (f x) * a) :=
(monotone_mul_right_of_nonneg ha).comp hf
```
as predicted (a is in a linearly ordered semiring)

Reply:
I guess one could really go through mathlib and try and make everything compatible, but I am not sure it's worth it. Then you'll get questions from people working on rings asking why `add_const` has no `'` but `mul_const'` does. I'm not sure there's a good solution here, but `to_additive` is powerful enough to handle what we have currently :D